File: test_policy.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from selenium.common.exceptions import TimeoutException

def run_test_policy():
    driver = webdriver.Chrome()
    driver.maximize_window()

    try:
        # Charger la page principale
        driver.get("https://connect.algoan.com/v2/init?client_id=030d0c7dfcfdcfcc135c6cf5&redirect_uri=https://dashboard.algoan.com")
        print("Test du lien 'Politique de confidentialité'...")

        try:
            # Localiser le lien 'Politique de confidentialité' et cliquer dessus
            policy_link = WebDriverWait(driver, 10).until(
                EC.element_to_be_clickable((By.LINK_TEXT, "Politique de confidentialité"))
            )
            policy_link.click()

            # Attendre que l'URL change pour correspondre à celle de la politique de confidentialité
            WebDriverWait(driver, 10).until(
                EC.url_contains("privacy-policy")
            )
            print("✅ Lien 'Politique de confidentialité' cliqué, redirection réussie.")

            # Retourner à la page principale
            driver.back()
            WebDriverWait(driver, 10).until(
                EC.url_contains("init")
            )
            print("✅ Retour à la page principale réussi.")
        except TimeoutException:
            print("❌ Test du lien 'Politique de confidentialité' échoué : la redirection ou le retour n'a pas fonctionné.")
    except Exception as e:
        print(f"Erreur inattendue : {e}")
    finally:
        driver.quit()

# Appeler la fonction pour exécuter le test
if __name__ == "__main__":
    run_test_policy()

File: test_policy.ts
import { chromium } from 'playwright';

export async function runTestPolicy() {
    const browser = await chromium.launch({ headless: false });
    const context = await browser.newContext();
    const page = await context.newPage();

    try {
        // Charger la page principale
        await page.goto("https://connect.algoan.com/v2/init?client_id=030d0c7dfcfdcfcc135c6cf5&redirect_uri=https://dashboard.algoan.com", { timeout: 10000 });
        console.log("Test du lien 'Politique de confidentialité'...");
        await page.waitForTimeout(2000);
        // Localiser le lien 'Politique de confidentialité' et cliquer dessus
        const policyLink = await page.waitForSelector("text=Politique de confidentialité", { timeout: 10000 });
        await page.waitForTimeout(2000);
        await policyLink.click();
        console.log("✅ Lien 'Politique de confidentialité' cliqué.");
        await page.waitForTimeout(2000);
        // Attendre que l'URL contienne "privacy-policy"
        await page.waitForURL(/privacy-policy/, { timeout: 10000 });
        console.log("✅ Redirection vers 'Politique de confidentialité' réussie.");
        await page.waitForTimeout(2000);
        // Vérifier que la page contient un élément clé (ex: titre h1)
        await page.waitForSelector("h1", { timeout: 10000 });
        console.log("✅ Contenu de la page Politique de confidentialité détecté.");
        await page.waitForTimeout(2000);
        // Retourner à la page principale
        await page.goBack();
        await page.waitForURL(/init/, { timeout: 10000 });
        console.log("✅ Retour à la page principale réussi.");
        await page.waitForTimeout(2000);
    } catch (error) {
        console.error("❌ Test du lien 'Politique de confidentialité' échoué :", error);
    } finally {
        await browser.close();
    }
}

if (require.main === module) {
    runTestPolicy();
}
